quiche/src/tls.rs

```diff
@@ -558,13 +554,21 @@ impl Handshake {
         map_result_ssl(self, rc)
     }
 
-    pub fn do_handshake(&mut self) -> Result<()> {
+    pub fn do_handshake(&mut self, conn_ptr: *mut Connection) -> Result<()> {
```

Comment:
You might be better off passing `&mut Connection` here and the other method rather than a raw pointer, to guarantee that connection is borrowed until the handshake is done.

Reply:
Due to how this method is called in `lib.rs` (`self.handshake.do_handshake(...)`), doing `self.handshake.do_handshake(self)` is considered double mutable reference, so using `&mut Connection` directly doesn't seem possible without some larger refactoring.

Reply:
Yeah, I was working on a separate change to address the issue you reported specifically, see https://github.com/cloudflare/quiche/pull/1231.